### PR TITLE
Updated file on lesson - 26th Oct Tutorial

### PR DESCRIPTION
**Pull request by nsahin**

### CODE REVIEW

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 1]
```diff
@@ -102,19 +99,17 @@
...
-    "- Displaying feature relationships, correlations\n",
-    "- Dimensionality reduction using PCA\n",
-    "- Scaling the data\n",
-    "- Classification and clustering algorithms\n",
-    "- Compare to another image dataset of handwritten digits"
+    "    -Displaying feature relationships, correlations\n",
+    "    -Dimensionality reduction using PCA\n",
+    "    -Scaling the data\n",
+    "    -Classification and clustering algorithms\n",
+    "    -Compare to another image dataset of handwritten digits"
```
Comment (joelostblom) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb]: These bullets will render correctly if you remove the indent before the `-` and add one space after the `-`.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 2]
```diff
@@ -126,33 +121,17 @@
...
-    "    corr = []\n",
-    "    for j in range(4):\n",
-    "        corr.append(stats.pearsonr(iris.data[:, i], iris.data[:, j])[0])\n",
-    "    feature_corr.loc[i, ] = corr\n",
-    "feature_corr = feature_corr.rename(index = {0: iris['feature_names'][0],\n",
-    "                                            1: iris['feature_names'][1],\n",
-    "                                            2: iris['feature_names'][2],\n",
-    "                                            3: iris['feature_names'][3]})\n",
-    "feature_corr"
+    "iris_df.iloc[:,:4].corr()"
```
Comment (joelostblom) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb:128]: Naming the column would make your intention more explicit:

```python
iris_df.drop('target', axis=1).corr()
```

**Review by joelostblom — CHANGES_REQUESTED** [review]
Thanks @nsahin!
Everything runs for me, just a couple of suggestions.

**Review by joelostblom — APPROVED** [review]

**Comment by joelostblom** [comment]
@nsahin Ping me when you are done updating and I will merge it.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 3]
```diff
@@ -79,13 +79,10 @@
...
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "collapsed": false
-   },
+   "metadata": {},
    "outputs": [],
    "source": [
     "%pylab inline\n",
-    "%matplotlib inline\n",
```
Comment (QuLogic) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb:88]: Should've dropped `%pylab` instead.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb (hunk 2 above)
Comment (QuLogic) [reply lessons/python/scikit-learn/UofTCoders_ML.ipynb:128]: OK, I was curious why you did it "by hand"; I guess these are equivalent then.

File: lessons/python/scikit-learn/UofTCoders_ML.ipynb [hunk 4]
```diff
@@ -328,30 +298,43 @@
     "pca = PCA(n_components=2)\n",
     "iris_pca_scaled = pca.fit_transform(X_scaled)\n",
     "\n",
-    "plt.figure(figsize=(6, 6))\n",
+    "plt.figure(figsize=(12, 6))\n",
+    "plt.subplot(121)\n",
     "plt.scatter(iris_pca_scaled[:, 0], iris_pca_scaled[:, 1], c=colors)\n",
```
Comment (QuLogic) [inline lessons/python/scikit-learn/UofTCoders_ML.ipynb:303]: Really wish you didn't use the global API so much though.

**Review by QuLogic — COMMENTED** [review]
I don't have any issues with this, but I do have some suggestions.
* It looks like you started from a very old version, which is causing conflicts right now. I would just throw away most of the old changes, but since it appears you are working on GitHub and not the command-line, I'm not sure how easy that is.
* Matplotlib's object-oriented interface could simplify a few things.
* [Seaborn](https://seaborn.pydata.org/) would also help remove a lot of the by-hand stuff, e.g., [the pair plots for feature relationships in one function call](https://seaborn.pydata.org/examples/scatterplot_matrix.html) or the [heatmap function for the confusion matrix](https://seaborn.pydata.org/examples/heatmap_annotation.html)

**Comment by joelostblom** [comment]
@nsahin If you want, I can help you updating this when we meet and discuss the other ML things.

@QuLogic What do you think of merging this as is and then I or @nsahin  will open a new PR addressing your comments and simplifying the plotting sections?

**Comment by QuLogic** [comment]
Again, those are just suggestions. It's probably better to leave it as it was presented for now.

**Pull request merged**